core/src/testils/criterion.rs

```diff
@@ -40,4 +49,47 @@ impl BencherExt for Bencher<'_> {
             total_elapsed
         })
     }
+
+    /// Similar to criterion's `iter_batched`, but uses a proptest strategy,
+    /// that returns an mbuf, as the setup to randomly generate a batch
+    /// inputs that can be polled for benchmarking pipeline-combinations and
+    /// operators.
+    fn iter_proptest_polled<R, S: Strategy<Value = Mbuf>, O: Batch>(
+        &mut self,
+        strategy: S,
+        mut routine: R,
+        batch_size: usize,
+    ) where
+        R: FnMut(Poll<Receiver<Mbuf>>) -> O,
+    {
+        self.iter_custom(|mut iters| {
+            let mut total_elapsed = Duration::from_secs(0);
+            let mut gen = Rvg::deterministic();
+            while iters > 0 {
+                let batch_size = cmp::min(batch_size, iters as usize);
+                let inputs = black_box(gen.generate_vec(&strategy, batch_size));
+                let mut outputs = Vec::with_capacity(batch_size);
+
+                let (mut tx, rx) = mpsc::channel();
+                tx.transmit(inputs.into_iter().collect::<Vec<_>>());
+                let mut new_batch = Poll::new(rx);
+                new_batch.replenish();
+
+                let start = Instant::now();
+                let mut batch = routine(new_batch);
+
+                while let Some(disp) = batch.next() {
+                    outputs.push(disp)
+                }
+
+                total_elapsed += start.elapsed();
+
+                black_box(batch);
```

Comment:
finally got curious and decided to read up on what [`black_box`](https://github.com/bheisler/criterion.rs/blob/master/book/src/faq.md#when-should-i-use-criterionblack_box) does. basically you only want to use it inside the timed portion of the bench. outside that, it's really pointless.

Reply:
so I used to remove this too, but https://docs.rs/criterion/0.3.0/src/criterion/lib.rs.html#518 showcases a black_box on outputs outside of the timer, no?

Reply:
yeah ok we will follow them.